internal/handler/mqtthandler/mqtt_handler.go

```diff
@@ -70,23 +69,47 @@ func NewHandler(server, username, password, cafile string) (handler.Handler, err
 	return &h, nil
 }
 
-func newTLSConfig(cafile string) (*tls.Config, error) {
+func newTLSConfig(cafile, certFile, certKeyFile string) (*tls.Config, error) {
+	if cafile == "" && (certFile == "" || certKeyFile == "") {
```

Comment:
I think there are three valid options:

* Only CA
* TLS cert + key
* CA, TLS + key

If in this case the CA cert is set + only the TLS cert or TLS key, it would pass this check, but the config is invalid I believe.